apps/examples/stack_protection/stack_protection_main.c

```diff
@@ -78,29 +83,69 @@ int stack_prot_main(int argc, char *argv[])
 #endif
 {
 	char ch;
+	char thread_type;
+	int tc_fd, ret_chk;
 	int i = 0;
+	bool type_kthread;
 #if defined(CONFIG_MPU_STACK_OVERFLOW_PROTECTION)
 	char *ptr;
 	char array_normal[STACK_NORMAL_SIZE];
 #endif
 
 	printf("Stack Overflow Protection testcase!!\n");
 
-	while (1) {
+	printf("\nPress U - Test Stack Protection for User Thread\n");
+	printf("Press K - Test Stack Protection for Kernel Thread\n");
+	printf("Press E - Exit the Test ..\n");
+	thread_type = getchar();
+
+	if (thread_type == 'K' || thread_type == 'k') {
+		tc_fd = open(TESTCASE_DRVPATH, O_WRONLY);
+		if (tc_fd < 0) {
+			printf("\nFailed to open testcase driver %s\n", TESTCASE_DRVPATH);
+			return ERROR;
+		}
+
+		type_kthread = true;
+		printf("\nTesting for Kernel Thread !!\n");
+	} else if (thread_type == 'U' || thread_type == 'u') {
+		type_kthread = false;
+		printf("\nTesting for User Thread !!\n");
+	} else {
+		printf(" Exit the Test ..\n");
+		return 0;
+	}
+
 #if defined(CONFIG_MPU_STACK_OVERFLOW_PROTECTION)
-		printf("\nPress M - Test Stack protection with MPU\n");
+	printf("\nTest Stack protection with MPU\n");
```

Comment:
Line 122 does not have '\n'. Let's keep the same format.
Add '\n' at 122 or remove it from this line

Reply:
Done